File: Examples/VisualBasic/Export/ExportImageToPSD.vb
Imports System.IO

Imports Aspose.Imaging

Namespace Aspose.Imaging.Examples.VisualBasic.Export
    Public Class ExportImageToPSD
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Export("ExportImageToPSD")

            'Load an existing image
            Using image As Image = image.Load(dataDir & "sample.bmp")
                'Create an instance of PsdOptions and set it’s various properties
                Dim psdOptions As New ImageOptions.PsdOptions()
                psdOptions.ColorMode = FileFormats.Psd.ColorModes.RGB
                psdOptions.CompressionMethod = FileFormats.Psd.CompressionMethod.RLE
                psdOptions.Version = 4

                'Save image to disk in PSD format
                image.Save(dataDir & "output.psd", psdOptions)

                ' Display Status.
                System.Console.WriteLine("Export to PSD performed successfully.")
            End Using
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ExportImageToPSD.cs
// GIST-ID: 6d6b95ddb9da89ca947752715c117d3b
using System;
using Aspose.Imaging.FileFormats.Psd;
using Aspose.Imaging.ImageOptions;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API
reference when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq
for more information. If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API
from https://releases.aspose.com/, install it, and then add its reference to this project. For any issues,
questions, or suggestions, please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    public class ExportImageToPSD
    {
        public static void Run()
        {
            Console.WriteLine("Running example ExportImageToPSD");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();
           
            // Load an existing image
            using (Image image = Image.Load(dataDir + "sample.bmp"))
            {
                // Create an instance of PsdOptions and set its various properties. Save the image to disk in PSD format.
                PsdOptions psdOptions = new PsdOptions();
                psdOptions.ColorMode = ColorModes.Rgb;
                psdOptions.CompressionMethod = CompressionMethod.Raw;
                psdOptions.Version = 4;
                image.Save(dataDir + "ExportImageToPSD_output.psd", psdOptions);               
            }

            Console.WriteLine("Finished example ExportImageToPSD");
        }
    }
}